### PR TITLE
Add golint as a tool and CI check

### PR DESCRIPTION
**Pull request by ifraixedes**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,3 +48,12 @@ else
 	@echo -e "error\tsome files did not pass go vet\n"
 	@go tool vet -all -shadow=true . 2>&1
 endif
+
+.PHONY: lintcheck
+lintcheck:
+ifeq ($(shell golint 2>&1 | wc -l), 0)
```
Comment (xescugc) [inline Makefile:54]: I would use this `golint -set_exit_status` instead of all this logic

**Review by xescugc — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (ifraixedes) [reply Makefile:54]: No idea how to make the same logic simpler with `golint -set_exist_status`

Could you tell me what you bear in mind, please?

This logic is the same applied in the other targets, so I wanted to reproduce the same behavior.

Comment (xescugc) [reply Makefile:54]: It's not exatly the same logic, it's a more "go" approch: Only print something is it goes wrong, if not, just stay quiet. I can't find the post where I read it, but it's basically what the `go get` does, what `dep` does and other "go" tools, except if the `-v` is setted.

But if this is not the approch of `raws` then go for the one you have instead.

Comment (ifraixedes) [reply Makefile:54]: I go for it, open an issue to change all of them at some point; I don't have the time to invest on it, right now, and I prefer to have `golint` the sooner although what there is in the makefile is not goish.

**Review by xescugc — APPROVED** [review]

**Pull request merged**